reconcile/utils/openshift_resource.py

```diff
@@ -501,6 +501,10 @@ def canonicalize(body):
             if spec.get("type") == "ClusterIP":
                 spec.pop("clusterIP", None)
 
+        if body["kind"] == "CronJob":
+            if body["apiVersion"] == "batch/v1beta1":
+                body["apiVersion"] = "batch/v1"
+
```

Comment:
I like the idea a lot. Can we add other resources listed in https://redhat:redhat@docs.openshift.com/container-platform/4.12/release_notes/ocp-4-12-release-notes.html#ocp-4-12-removed-kube-1-25-apis ? Especially:


| Resource | removed APIGroup | correct APIGroup
|----------|---------------------|-----------------
| HorizontalPodAutoscaler | autoscaling/v2beta1 | autoscaling/v2
| PodDisruptionBudget | policy/v1beta1 | policy/v1

Reply:
sounds like a ticket instead of this small PR :)